src/Coreutils-EYAD_HUSSEIN/pkg/tree.go

```diff
@@ -0,0 +1,95 @@
+package pkg
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func Tree() {
+
+	var maxDepth int
+
+	flag.IntVar(&maxDepth, "L", -1, "Descend only level directories deep")
```

Comment:
why -1?

Reply:
this way, given no flag, the function will never return from the first base case, which is

```
if maxDepth != -1 && depth > maxDepth {
		return
	}
```

This way, the recursion tree won't stop until every file and directory in each level inside whatever entry point is given is printed(expect for hidden ones of course(second base case)) --> all levels are traversed and printed, which is the behavior of tree when no flag is passed.